frontend/app/(auth)/register.tsx

```diff
@@ -0,0 +1,160 @@
+import UserInput from '@/components/UserInput';
+import { RegisterSchema, RegisterSchemaInputs } from '@/types/user';
+import { isClerkAPIResponseError, useSignUp } from '@clerk/clerk-expo';
+import { SignUpStatus } from '@clerk/types';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useRouter } from 'expo-router';
+import { useState } from 'react';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { Button, Form, Separator, Text, View, XStack, YStack } from 'tamagui';
+
+const Register = () => {
+  const {
+    signUp: { create },
+    setActive,
+  } = useSignUp();
+
+  const router = useRouter();
+  const [status, setStatus] = useState<SignUpStatus>();
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<RegisterSchemaInputs>({
+    resolver: zodResolver(RegisterSchema),
+    defaultValues: {
+      username: '',
+      password: '',
+    },
+  });
+  const registerUser: SubmitHandler<RegisterSchemaInputs> = async (data) => {
+    try {
+      const { createdSessionId, status } = await create({
+        username: data.username,
+        password: data.password,
+        firstName: data.firstName ?? '',
+        lastName: data.lastName ?? '',
+      });
+
```

Comment:
The register user works well. The checks for the password are a good implementation!